### PR TITLE
Adding sync up script to add data here

### PR DESCRIPTION
**Pull request by vdeturckheim**

### CODE REVIEW

File: tools/sync_up.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+// collect and set up ecosystem reports
+fs.readdirSync(path.join(SWG_ROOT, 'npm'))
+    .forEach((vulnname) => {
+
+        const rawContent = fs.readFileSync(path.join(SWG_ROOT, 'npm', vulnname));
+        const content = JSON.parse(rawContent);
+        const pkg = content.module_name;
+
+        const targetPath = path.join(NPM_TARGET_ROOT, pkg);
+        if(!fs.existsSync(targetPath)) {
```
Comment (thefourtheye) [inline tools/sync_up.js:18]: We shouldn't check before creating, right?

File: tools/sync_up.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+            fs.mkdirSync(targetPath);
+        }
+        if (!fs.existsSync(path.join(targetPath, vulnname))) {
```
Comment (thefourtheye) [inline tools/sync_up.js]: Is the intention not to overwrite the file?

File: tools/sync_up.js (hunk 1 above)
Comment (vdeturckheim) [reply tools/sync_up.js:18]: I'm not sure I get this comment :/

File: tools/sync_up.js (hunk 2 above)
Comment (vdeturckheim) [reply tools/sync_up.js]: Yes, but thinking of it, we should indeed override the file

File: tools/sync_up.js (hunk 1 above)
Comment (thefourtheye) [reply tools/sync_up.js:18]: Sorry, I should have explained myself better there. Before creating a file or directory we should not check if it exists. We should just try to create. If it fails, it's already there. That would be more reliable, I believe.

File: tools/sync_up.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+const fs = require('fs');
+const path = require('path');
+
+const SWG_ROOT = path.join(__dirname, '..', 'node_modules', 'security-wg', 'vuln');
+const NPM_TARGET_ROOT = path.join(__dirname, '..', 'ecosystem');
+const CORE_TARGET_ROOT = path.join(__dirname, '..', 'core');
+
+// collect and set up ecosystem reports
+fs.readdirSync(path.join(SWG_ROOT, 'npm'))
+    .forEach((vulnname) => {
```
Comment (cjihrig) [inline tools/sync_up.js]: This fits on the previous line and is still under 80 characters.

File: tools/sync_up.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
+'use strict';
+const fs = require('fs');
+const path = require('path');
+
+const SWG_ROOT = path.join(__dirname, '..', 'node_modules', 'security-wg', 'vuln');
+const NPM_TARGET_ROOT = path.join(__dirname, '..', 'ecosystem');
+const CORE_TARGET_ROOT = path.join(__dirname, '..', 'core');
+
+// collect and set up ecosystem reports
+fs.readdirSync(path.join(SWG_ROOT, 'npm'))
```
Comment (cjihrig) [inline tools/sync_up.js]: Maybe store this `join()` result in a variable since it's also part of the `join()` below.

File: tools/sync_up.js [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    .forEach((vulnname) => {
+
+        const rawContent = fs.readFileSync(path.join(SWG_ROOT, 'npm', vulnname));
+        const content = JSON.parse(rawContent);
```
Comment (cjihrig) [inline tools/sync_up.js:14]: This can throw, but I guess we should probably crash in that case anyway.

File: tools/sync_up.js [hunk 6]
```diff
@@ -0,0 +1,35 @@
+'use strict';
+const fs = require('fs');
+const path = require('path');
+
+const SWG_ROOT = path.join(__dirname, '..', 'node_modules', 'security-wg', 'vuln');
```
Comment (lirantal) [inline tools/sync_up.js]: can we rename the var to make it easy to understand that this is actually the current original repo? something like maybe `SWG_ROOT_SRC` or something else.

File: tools/sync_up.js (hunk 3 above)
Comment (lirantal) [reply tools/sync_up.js]: Colin are you editing on tmux? ;-)

File: tools/sync_up.js [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+        const pkg = content.module_name;
+
+        const targetPath = path.join(NPM_TARGET_ROOT, pkg);
+        if(!fs.existsSync(targetPath)) {
+            fs.mkdirSync(targetPath);
+        }
+        fs.writeFileSync(path.join(targetPath, vulnname), rawContent);
+    });
+
+// same for core vulns
```
Comment (lirantal) [inline tools/sync_up.js:24]: we could probably have just one function that does this and we give it different dirs as source and target so we keep the code simpler and smaller.

**Review by lirantal — COMMENTED** [review]
LG, left a few comments

File: tools/sync_up.js (hunk 1 above)
Comment (vdeturckheim) [reply tools/sync_up.js:18]: I am not sure, I might have to add vulnerabilities to a dir already existing

File: tools/sync_up.js (hunk 7 above)
Comment (vdeturckheim) [reply tools/sync_up.js:24]: Not sure; the logic is not fully similar here (create subdirs or not)

**Pull request merged**